peer.go

```diff
@@ -174,9 +174,20 @@ type peer struct {
 	wg        sync.WaitGroup
 }
 
+type PeerCreator interface {
```

Comment:
Missing a `godoc` comment here. 

Also not really seeing the requirement for an interface like this. Why can't the `newPeer` function be used directly?

Reply:
Using the newPeer function for unit testing is a headache. Figuring out how to make the Decrypt function not fail is very non-trivial. That said, I'm trying to do that now. But it doesn't seem like the unit testing for server should be so strongly coupled to peer. Making the PeerCreator interface was an attempt to separate these unit testing concerns.

Reply:
I removed the PeerCreator interface and instead abstracted out the database calls, and stubbed out more server functionality.